Add ClickSend "caller" option

fabtesta opened the pull request:
Added optional "caller" parameter to let ClickSend API call be more compliant with some cellular networks that do not allow incoming calls from the same recipient number.
If "caller" is not defined, then the recipient parameter value is used.

**Doc PR**
https://github.com/home-assistant/home-assistant.io/pull/8831

fabtesta committed:
removed "from" parameter from ClickSend API call

Removed "from" parameter from API call to let ClickSend choose a random number and be more compliant with some cellular networks that do not allow incoming calls from the same recipient number.

homeassistant commented:
Hi @fabtesta, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#20780).

Once you do that we will be able to review and accept this pull request.

Thanks!

fabtesta committed:
fixed "line too long (94 > 79 characters)" houndci code review

fixed "line too long (94 > 79 characters)" houndci code review

bachya commented:
Base this PR upon `home-assistant:dev`, not `home-assistant:master`.

balloob reviewed (changes requested):
The default logic should still keep using the recipient. Adjust this PR to make it optional.

fabtesta committed:
Added new component optional parameter "caller".
If defined is used to override default logic that uses recipient phone number in "from" API call parameter

MartinHjelmare commented on homeassistant/components/notify/clicksend_tts.py:
```diff
@@ -38,6 +40,7 @@
     vol.Required(CONF_RECIPIENT): cv.string,
     vol.Optional(CONF_LANGUAGE, default=DEFAULT_LANGUAGE): cv.string,
     vol.Optional(CONF_VOICE, default=DEFAULT_VOICE): cv.string,
+    vol.Optional(CONF_CALLER, default=DEFAULT_CALLER): cv.string,
```
Remove default. We will check for `None` below.

MartinHjelmare commented on homeassistant/components/notify/clicksend_tts.py:
```diff
@@ -60,10 +63,13 @@ def __init__(self, config):
         self.language = config.get(CONF_LANGUAGE)
         self.voice = config.get(CONF_VOICE)
+        self.caller = config.get(CONF_CALLER)
+        if self.caller == '':
```
```py
if self.caller is None:
```

MartinHjelmare commented:
Please update the PR description to follow the change in PR direction.

fabtesta committed:
Removed default value for CALLER parameter. If not defined then will take the RECIPIENT value.

MartinHjelmare commented:
Can be merged when PR description is updated and a docs PR is linked in the PR description.
